contracts/contracts/UserReplicaSetManager.sol

```diff
@@ -0,0 +1,393 @@
+pragma solidity ^0.5.0;
+
+import "./interface/RegistryInterface.sol";
+import "./registry/RegistryContract.sol";
+import "./UserFactory.sol";
+
+import "./SigningLogicInitializable.sol";
+
+
+/** @title Contract for Audius user replica set management */
+contract UserReplicaSetManager is SigningLogicInitializable, RegistryContract {
+    /// @notice Reference to data contract registry
+    RegistryInterface private registry = RegistryInterface(0);
+    /// @notice User factory key used to confirm valid users during reconfig operations
+    bytes32 private userFactoryRegistryKey;
+
+    /// @notice Address permissioned to update user replica sets
+    address userReplicaSetBootstrapAddress;
+
+    /// @notice ServiceProvider ID to ServiceProvider delegateOwnerWallet,
+    ///     reflecting registered values on Audius Ethereum L1 contracts
+    mapping (uint => address) spIdToContentNodeDelegateWallet;
+
+    /// @notice Struct used to represent replica sets
+    //          Each uint represents the spID registered on eth-contracts
+    struct ReplicaSet {
+        uint primary;
+        uint[] secondaries;
+    }
+
+    /// @notice Current uint userId to Replica Set
+    mapping (uint => ReplicaSet) userReplicaSets;
+
+    /* Events */
+    event UpdateReplicaSet(
```

Comment:
add `indexed` field here

Reply:
i think we actually don't need an `indexed` field here - will add details

Reply:
since indexed fields are used to query events through web3 filters and we have our own set of indexed data in the discovery-node corresponding to every event emission over time, adding this keyword is unnecessary - it also has an additional implication to the indexing flow itself, as these fields require an entirely new contract/event interaction. given the overhead and relative lack of benefit we have opted to bypass the indexed keyword entirely